### PR TITLE
Improve url for better bookmarking

### PR DESCRIPTION
**Pull request by oorion**
Update the url path upon searching for an address, block hash, or transaction id.  Forward and backward navigation should also work.  A user would be able to bookmark a page properly with these changes.  

Other than adding specs, I'd like to know if there are any issues with the direction I'm going.

### CODE REVIEW

**Comment by zquestz** [comment]
Love the direction! Going to test this out now, this was on my todo list! Thanks!

**Comment by zquestz** [comment]
Just brought up your branch and the changes work great. Going forward and back in the browser works correctly, and bookmarks are possible. Thanks so much. If you want to add some tests that would be great, but admittedly the test harness is VERY light as the gRPC library doesn't seem t o. work when running in the tests...

**Comment by zquestz** [comment]
Minor nit, when someone submits an empty search, right now we clear the results and we should probably update the url in that case. =)

**Comment by zquestz** [comment]
Also, right now it doesn't respect mainnet/testnet... not sure the best way to handle that, but we should probably make sure that is also functional... maybe a network param in the url? `/mainnet/blocks` and `/testnet/blocks`?

**Comment by oorion** [comment]
I'll see if I can mock the gRPC when I write some specs.  I'll also make sure empty searches reset the url.  

`/mainnet/block` and `/testnet/block` is a good approach in my opinion.  I'll get that working as well.

File: src/main.js [hunk 1]
```diff
@@ -16,9 +16,26 @@
...
 Vue.config.productionTip = false;
 
+Vue.use(VueRouter)
+const router = new VueRouter({
+  mode: 'history',
+  routes: [
+    {name: 'home', path: '/', component: BCHDExplorer},
+    {name: 'mainnet/address', path: '/mainnet/address/:address', component: BCHDExplorer},
+    {name: 'testnet/address', path: '/testnet/address/:address', component: BCHDExplorer},
+    {name: 'mainnet/block', path: '/mainnet/block/:blockHash', component: BCHDExplorer},
```
Comment (zquestz) [inline src/main.js:31]: Can't we do something like `path: '/:network/address/:address`? Then we can cut the routes in half?

File: src/components/BCHDExplorer.vue [hunk 2]
```diff
@@ -89,32 +89,59 @@ export default {
...
+    } else if (params.blockHash) {
+      this.searchBCHD(params.blockHash)
+    } else if (params.txId) {
+      this.searchBCHD(params.txId)
+    }
+  },
+  watch: {
+    $route(to) {
+      this.input = ""
+      const input = to.params.address || to.params.blockHash || to.params.txId
```
Comment (zquestz) [inline src/components/BCHDExplorer.vue:105]: Here we should be able to look up `to.params.network` then make sure the network is set correctly and "tick" then testnet checkbox if necessary.

File: src/main.js (hunk 1 above)
Comment (oorion) [reply src/main.js:31]: Makes sense.  Thanks for the feedback.  This is my first time working in vue.

File: src/components/BCHDExplorer.vue (hunk 2 above)
Comment (oorion) [reply src/components/BCHDExplorer.vue:105]: Yep, I had done that in a commit I hadn't pushed until just now.  Will update it to use `to.params.network`.  I spent some time this weekend trying to mock the rpc library but didn't have much success with jest.

Comment (zquestz) [reply src/components/BCHDExplorer.vue:105]: I had the same problems. We can merge these changes without the mocks. Finding a good testing pattern might take some time.

File: src/main.js (hunk 1 above)
Comment (zquestz) [reply src/main.js:31]: It was my first forray into Vue as well!

**Comment by oorion** [comment]
I'll work on specs on a separate branch.

**Comment by zquestz** [comment]
Will test your changes today. Looks good from reading the diff. :-)

**Comment by heimdall-asguard** [comment]
This Pull Request was merged without enough reviews
Please go [here](https://github.com/coinbase/bchd-explorer/issues/5) to review and +1

**Pull request merged**